modules/authorization-xacml/src/main/java/org/opencastproject/authorization/xacml/XACMLAuthorizationService.java

```diff
@@ -488,6 +485,7 @@ public Boolean apply(Attachment attachment) {
             }
 
             PolicyDecisionPoint pdp = getPolicyDecisionPoint(xacmlPolicyFile);
+            FileUtils.deleteQuietly(xacmlPolicyFile);
```

Comment:
Try to not leave files in the workspace

Reply:
I suggest to move that line into the method getPolicyDicisionPoint so that all file handling stuff lies within that method.

What do you think?

Reply:
As ultimate the file contents are converted to a string inside PolicyDecisionPoint getPolicyDecisionPoint(File xacmlFile), we could move that conversion higher up and use workspace.read() and convert inputstream > string, then call getPolicyDecisionPoint(String acl) ?

Reply:
Yes, this module would benefit from some refactoring. Unfortunately, I have no time for that right now. This patch is just a minor fix to get a maximum effect with minimal effort. I cannot spend more time on this at the moment.

@staubesv, closing the file is done in the same method as getting/opening the file is done. I think opening the file early was done to be able to already fail at that point and not do any unnecessary work. Again, refactoring this might be beneficial. based on the old code, I think it makes sense to clode the file here.

Reply:
@lkiesow Oh, you're right. Haven't looked at the code enough...